### PR TITLE
#44499 add to playlist action and misc changes

### PR DESCRIPTION
**Pull request by manneohrstrom**
- added an add to playlist action to versions, making it easy to add versions to playlists:

screen shot 2018-01-09 at 10 35 19
screen shot 2018-01-09 at 03 10 45

- fixed some code comments
- project field now more generally available in hooks
- updated to use shotgunmenu and support grouped actions:

screen shot 2018-01-09 at 10 37 08

### CODE REVIEW

File: hooks/general_actions.py [hunk 1]
```diff
@@ -87,9 +92,59 @@ def generate_actions(self, sg_data, actions, ui_area):
...
+                        ]
+                    }
+                ],
+                ["code", "id", "sg_date_and_time"],
+                order=[{"field_name": "updated_at", "direction": "desc"}],
+                limit=10,
+            )
+
+            # playlists this version is already part of
+            existing_playlist_ids = [x["id"] for x in (sg_data.get("playlists") or [])]
```
Comment (josh-t) [inline hooks/general_actions.py]: nitpick: can you do `sg_data.get("playlists", [])` instead?

File: hooks/general_actions.py [hunk 2]
```diff
@@ -116,7 +171,15 @@ def execute_action(self, name, params, sg_data):
             assignees.append(app.context.user)
             app.shotgun.update("Task", sg_data["id"], {"task_assignees": assignees})
 
-        elif name == "task_to_ip":        
+        elif name == "add_to_playlist":
+            app.shotgun.update(
```
Comment (josh-t) [inline hooks/general_actions.py]: Might be good to throw a debug statement in here.

File: hooks/general_actions.py [hunk 3]
```diff
@@ -87,9 +92,59 @@ def generate_actions(self, sg_data, actions, ui_area):
...
+
+                if playlist.get("sg_date_and_time"):
+                    # playlist name includes date/time
+                    caption = "%s (%s)" % (
+                        playlist["code"],
+                        self._format_timestamp(playlist["sg_date_and_time"])
+                    )
+                else:
+                    caption = playlist["code"]
+
```
Comment (josh-t) [inline hooks/general_actions.py]: Suggest a debug statement in here.

File: hooks/general_actions.py (hunk 1 above)
Comment (manneohrstrom) [reply hooks/general_actions.py]: yep will fix!

File: hooks/general_actions.py (hunk 3 above)
Comment (manneohrstrom) [reply hooks/general_actions.py]: will add!

File: hooks/general_actions.py (hunk 2 above)
Comment (manneohrstrom) [reply hooks/general_actions.py]: will add!

**Pull request merged**